pennylane_lightning/core/src/simulators/lightning_qubit/catalyst/LightningSimulator.cpp

```diff
@@ -45,14 +66,11 @@ auto LightningSimulator::AllocateQubits(size_t num_qubits)
 }
 
 void LightningSimulator::ReleaseAllQubits() {
-    this->device_sv->clearData();
     this->qubit_manager.ReleaseAll();
+    this->device_sv = std::make_unique<StateVectorT>(0); // reset the device
 }
 
 void LightningSimulator::ReleaseQubit(QubitIdType q) {
-    if (this->qubit_manager.isValidQubitId(q)) {
-        this->device_sv->releaseWire(this->qubit_manager.getDeviceId(q));
-    }
```

Comment:
If the answer to the above is yes, does the removal of this block break things, or will the statevector merely carry around a "logically dead" qubit that will never be addressed again but still needs to be updated by gate kernels?

Reply:
@dime10 I don't understand your question.